app/scripts/components/exploration/components/dataset-selector-modal/content.tsx

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { glsp, themeVal, media } from '@devseed-ui/theme-provider';
+import {
+  CollecticonPlus,
+  CollecticonTickSmall,
+  iconDataURI
+} from '@devseed-ui/collecticons';
+import { DatasetLayerCardProps } from './';
+
+import { DatasetClassification } from '$components/common/dataset-classification';
+import EmptyHub from '$components/common/empty-hub';
+import {
+  Card,
+  CardList,
+  CardMeta,
+  CardTopicsList
+} from '$components/common/card';
+import TextHighlight from '$components/common/text-highlight';
+import { CardSourcesList } from '$components/common/card-sources';
+import { getDatasetPath } from '$utils/routes';
+import {
+  getTaxonomy,
+  TAXONOMY_SOURCE,
+  TAXONOMY_TOPICS
+} from '$utils/veda-data';
+import { Pill } from '$styles/pill';
+
+const DatasetContainer = styled.div`
+  height: auto;
+  display: flex;
+  margin-bottom: ${glsp(2)};
+
+  ${CardList} {
+    width: 100%;
+  }
+
+  ${EmptyHub} {
+    flex-grow: 1;
+  }
+`;
+
+const SingleDataset = styled.div`
+  padding-bottom : ${glsp(2)};
+  &:not(:last-child) {
+    border-bottom : 1px solid ${themeVal('color.base-200')};
+  }
+`;
+
+const DatasetSelectedLayer = styled.div`
+  background-color: ${themeVal('color.primary')};
+  border-radius: ${themeVal('shape.ellipsoid')};
+  padding: 0 ${glsp(0.5)};
+  color: ${themeVal('color.surface')};
+`;
+
+const DatasetHeadline = styled.div`
+  display: flex;
+  gap: ${glsp(1)};
+  margin-bottom: ${glsp(1)};
+`;
+const DatasetDescription = styled.p``;
+const DatasetIntro = styled.div`
+  padding: ${glsp(1)} 0;
+  ${DatasetDescription} {
+    ${media.largeUp`
+      width: 66%;
+    `}
+  }
+`;
+
+interface ModalContentComponentProps {
+  search: string;
+  selectedIds: string[];
+  displayDatasets: any[];
+  onCheck:  (id:any) => void;
+}
+
+export default function ModalContentComponent(props:ModalContentComponentProps) {
+  const { search, selectedIds, displayDatasets, onCheck } = props;
+  return(<>
+  <DatasetContainer>
+    {displayDatasets.length ? (
+      <div>
+      {displayDatasets.map(currentDataset => (
+        <SingleDataset key={currentDataset.id}>
+          <DatasetIntro>
+            <DatasetHeadline>
+            <h4>{currentDataset.name}</h4>
+            {/* <Subtitle><LayerInfoLiner info={currentDataset.subtitle}/></Subtitle> */}
+            {currentDataset.countSelectedLayers > 0 && <DatasetSelectedLayer><span>{currentDataset.countSelectedLayers} selected </span> </DatasetSelectedLayer>}
```

Comment:
❓ : There is commented code sprinkled through out, im guessing this is still kindof WIP? we should get rid of these before merging ~

🔧 : typing `displayDatasets` would allow us to type `countSelectedLayers` to `number` and then we can just do `{currentDataset.countSelectedLayers  && ...` instead

Reply:
typing is a great idea, but `currentDataset.countSelectedLayers > 0 &&` seems to be still necessary; otherwise, `0` will be rendered.